web/src/engine/predictive-text/worker-thread/src/main/correction/classical-calculation.ts

```diff
@@ -272,78 +324,210 @@ export class ClassicalDistanceCalculation<TUnit = string, TInput extends EditTok
     return false;
   }
 
+  /**
+   * Determines the edit path used to obtain the optimal cost, distinguishing between zero-cost
+   * substitutions ('match' operations) and actual substitutions.
+   *
+   * If multiple edit paths are valid, this method will also sort them in a manner favoring, in order:
+   * - long, contiguous stretches of 'match' entries
+   * - long, contiguous stretches of 'match' + 'substitute' entries
+   * - long, contiguous stretches of 'match' + 'substitute' + 'transpose' entries
+   * @param row
+   * @param col
+   */
+  public editPath(): EditTuple<TUnit>[][] {
```

Comment:
An idea I toyed with was to pass in a function to validate substitutions - allowing us to use `isSubstitutionAlignable` for validation of full-word token substitutions and ensure that we prefer paths where all substitutions "look good".  I think we actually do get the results we need without having to invoke this approach, which is something I'm pleased about here.

It would still be possible to add it as a parameter to this outer method for use in path ranking in the future should the need arise.  ("All valid substitutions" is preferable over "some invalid substitutions", etc.)